src/dotnet/commands/dotnet-new/Program.cs

```diff
@@ -37,32 +37,30 @@ where resourceName.Contains(templateDir)
             {
                 string fileName = GetFileNameFromResourceName(resourceName);
 
-                using (var fileStream = File.Create(fileName))
+                using (var resource = thisAssembly.GetManifestResourceStream(resourceName))
                 {
-                    using (var resource = thisAssembly.GetManifestResourceStream(resourceName))
-                    {
-                        resource.CopyTo(fileStream);
-                    }
-                }
+                    var archive = new ZipArchive(resource);
 
-                try
-                {
-                    ZipFile.ExtractToDirectory(fileName, Directory.GetCurrentDirectory());
+                    try
+                    {
+                        archive.ExtractToDirectory(Directory.GetCurrentDirectory());
 
-                    File.Move(
-                        Path.Combine(Directory.GetCurrentDirectory(), "project.json.template"),
-                        Path.Combine(Directory.GetCurrentDirectory(), "project.json"));
-                }
-                catch (IOException ex)
-                {
-                    Reporter.Error.WriteLine(ex.Message);
-                    hasFilesToOverride = true;
-                }
-                finally
-                {
-                    if (File.Exists(fileName))
+                        File.Move(
+                            Path.Combine(Directory.GetCurrentDirectory(), "project.json.template"),
+                            Path.Combine(Directory.GetCurrentDirectory(), "project.json"));
+                    }
+                    catch (IOException ex)
+                    {
+                        Reporter.Error.WriteLine(ex.Message);
+                        hasFilesToOverride = true;
+                    }
+                    finally
                     {
-                        File.Delete(fileName);
+                        var noautoupdate = Path.Combine(Directory.GetCurrentDirectory(), ".noautoupdate");
```

Comment:
I think it would be best if this ".noautoupdate" file didn't exist in the .zip file at all.

Maybe it would be best to just exclude this directory in the update-dependencies script?